services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaService.kt

```diff
@@ -67,19 +63,21 @@ class HankeKayttajaService(
         )
     }
 
-    private fun createInvitationToken(hankeId: Int, contact: UserContact) {
+    private fun createToken(hankeId: Int, contact: UserContact) {
         logger.info { "Creating a new user token, hankeId=$hankeId" }
         val token = KayttajaTunnisteEntity.create()
         val kayttajaTunnisteEntity = kayttajaTunnisteRepository.save(token)
         logger.info { "Saved the new user token, id=${kayttajaTunnisteEntity.id}" }
 
         saveUser(
-            HankeKayttajaEntity(
-                hankeId = hankeId,
-                nimi = contact.name,
-                sahkoposti = contact.email,
-                permission = null,
-                kayttajaTunniste = kayttajaTunnisteEntity
+            hankeKayttajaRepository.save(
```

Comment:
`saveUser` has `hankeKayttajaRepository.save(hankeKayttajaEntity)` as it's first expression, so adding it here is quite redundant.

Reply:
There is a save operation and a log operation. I'm not sure whether the function is redundant or not.

Reply:
I don't think the saveUser is redundant, saving the entity here before passing it to saveUser is.

Reply:
Oops, yes now I see the issue.

Reply:
Fixed.